### PR TITLE
feat: add a GitHub action

### PR DESCRIPTION
**Pull request by lucacasonato**

### CODE REVIEW

**Review by satyarohith — COMMENTED** [review]
Maybe we should avoid checking in node_modules? 

We have the lock file and makefile, so might a build command be good?

**Comment by lucacasonato** [comment]
> Maybe we should avoid checking in node_modules?

I thought this is needed on GHA. I guess we could bundle all the code together, but I don't really like bundling backend code. This is something we could address in a follow up.

File: action/index.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+      return;
+    }
+  }
+
+  const aud = new URL(`/projects/${projectId}`, ORIGIN);
+  let token;
+  try {
+    token = await core.getIDToken(aud);
+  } catch {
+    throw "Failed to get the GitHub OIDC token. Make sure that this job has the required permissions for getting GitHub OIDC tokens (see https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/about-security-hardening-with-openid-connect#adding-permissions-settings ).";
```
Comment (satyarohith) [inline action/index.js:35]:
```suggestion
    throw "Failed to get the GitHub OIDC token. Make sure that this job has the required permissions for getting GitHub OIDC tokens (see https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/about-security-hardening-with-openid-connect#adding-permissions-settings).";
```
nit

File: action/README.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+To deploy you just need to include the Deno Deploy GitHub Action as a step in
+your workflow.
+
+You do **not** need to set up any secrets for this to work. You **do** need to
+link your GitHub repository to your Deno Deploy project and choose the "GitHub
+Actions" deployment mode. You can do this in your project settings on
+https://dash.deno.com.
+
+```yml
+- name: Deploy to Deno Deploy
```
Comment (satyarohith) [inline action/README.md:22]: We should add a note about `permissions` for the step? Or include it in the setup yaml.

**Review by satyarohith — COMMENTED** [review]
LGTM.

**Pull request merged**